### PR TITLE
User Stories 10 - 12 ZTH

### PR DESCRIPTION
**Pull request by zachholcomb**
Complete user stories 10-12
Adds user create functionality
Adds flash message for incomplete user forms
- [x]  Passing all tests

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      redirect_to "/profile"
+    else
+      flash[:notice] = @user.errors.full_messages.to_sentence
+      render :new
+    end
+  end
+
+  def show
+  end
+
```
Comment (steve-meyers) [inline app/controllers/users_controller.rb:19]: Do we need this show method at this point?

Comment (zachholcomb) [reply app/controllers/users_controller.rb:19]: Yes profile leads us here from the routes file

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<h1>New User Registration</h1>
+<p>Enter details below</p>
```
Comment (steve-meyers) [inline app/views/users/new.html.erb]: This line is self-explanatory and I think we should consider removing it during a refactor. Thoughts?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.create(user_params)
+    if @user.save
+      flash[:notice] = "You are successfully registered and logged in!"
+      redirect_to "/profile"
+    else
+      flash[:notice] = @user.errors.full_messages.to_sentence
```
Comment (elguapogordo) [inline app/controllers/users_controller.rb:12]: so happy to have discovered this!

File: app/views/users/new.html.erb (hunk 2 above)
Comment (zachholcomb) [reply app/views/users/new.html.erb]: I agree, not necessary

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (elguapogordo) [reply app/controllers/users_controller.rb:19]: it self identifies as an action, not a method

File: app/views/users/new.html.erb (hunk 2 above)
Comment (elguapogordo) [reply app/views/users/new.html.erb]: also agree. "below" consists of the entire page, which is just one form.

File: spec/features/users/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    fill_in :City, with: ""
+    fill_in :State, with: "GA"
+    fill_in :Zip, with: "66666"
+    fill_in :Email, with: "chunky_lover@example.com"
+    fill_in :Password, with: "123password"
+    fill_in :user_password_confirmation, with: "123password"
+    click_button "Create User"
+    expect(page).to have_content("Name can't be blank and City can't be blank")
+  end
+
```
Comment (steve-meyers) [inline spec/features/users/create_spec.rb:32]: Should we add additional edge case testing here?

File: spec/features/users/create_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  it "cannot be created with an email that already exists" do
+    user1 = User.create!(name: "Steve",
+                        address:"123 Main St.",
+                        city: "Fort Collins",
+                        state: "GA",
+                        zip: "66666",
+                        email: "chunky_lover@example.com",
+                        password: "123password")
```
Comment (steve-meyers) [inline spec/features/users/create_spec.rb:40]: This is a very insecure password.

**Review by steve-meyers — APPROVED** [review]
Everything looks good but we should remember to go back and add some additional edge-case testing. We should decide on how thoroughly we should screen the incoming form data.

File: app/views/users/new.html.erb (hunk 2 above)
Comment (jrsewell400) [reply app/views/users/new.html.erb]: I removed line 2

**Comment by jrsewell400** [comment]
Reviewed changes, looks good!

**Pull request merged**